cilium-health/launch/endpoint.go

```diff
@@ -42,6 +44,20 @@ import (
 	"github.com/vishvananda/netlink"
 )
 
+var (
+	// vethName is the host-side link device name for cilium-health EP.
+	vethName = "cilium_health"
+
+	// vethPeerName is the endpoint-side link device name for cilium-health.
+	vethPeerName = "cilium"
+
+	// healthPidfile
+	healthPidfile = "health-endpoint.pid"
+
+	// client is used to ping the cilium-health endpoint as a health check.
+	client *healthPkg.Client
```

Comment:
I don't see an easy way to do this better, but this will be slightly racey between callers of `PingEndpoint` and setting this in `LaunchAsEndpoint`. The only other way I see is to make the client a parameter here and have callers worry about creating it at the correct time.

Reply:
That's why the ping endpoint returns an error if the client is nil, it always hits this the very first time with the current code. I thought about adding a struct to manage the lifetime of this client and let the caller use it and so on, that's probably the most idiomatic way to handle this. Maybe I should just go ahead and do that.

Reply:
Oh, I mean it would trigger the race detector, and arbitrarily fail that. I suppose it's possible to get a partial read of the variable but I'm pretty sure that won't happen.

How important is it to only have one of these? I imagine the overhead of a few is pretty low. In any case a sync.Once to manage the global singleton would work, but there's a really annoying sequencing requirement here, since I'm guessing it will fail to create if the health endpoint is down, and would need to be recreated on errors?

Reply:
Oh, I think it's only ever touched from the controller goroutine right now. But that means it's not a thread-safe API.

We only need one of them right now, the thing I was trying to avoid is creating a new connection for every request. Yes, it will block and time out if the endpoint is down - which is also why it seemed like a good way to choose when to recreate the endpoint.